tests/test_measurements.py

```diff
@@ -867,6 +867,70 @@ def circuit():
         for ind in counts_term_indices:
             assert isinstance(res[ind], dict)
 
+    def test_all_outcomes_kwarg_providing_observable(self):
```

Comment:
Let's also have a test case where we have multiple measurements including `qml.counts(..., all_outcomes=True)`.